app/models/user.rb

```diff
@@ -89,14 +89,17 @@ def no_attempt_for_two_weeks
     # Get ACTIVE volunteers that have ACTIVE supervisor assignments with at least one ACTIVE case
     # 1st condition: Volunteer has not created a contact AT ALL within the past 14 days
 
-    volunteers
-      .includes(:case_assignments)
-      .joins("LEFT JOIN case_contacts cc on cc.creator_id = users.id AND cc.occurred_at > (CURRENT_DATE - INTERVAL '14 days')")
-      .group("users.id, supervisor_volunteers_users.id, case_assignments.id")
-      .where(active: true)
-      .where(supervisor_volunteers: {is_active: true})
-      .where(case_assignments: {active: true})
-      .length
+    no_attempt_count = 0
+    volunteers.map do |volunteer|
+      if volunteer.supervisor.active? &&
+          volunteer.case_assignments.any? { |assignment| assignment.active? } &&
```

Comment:
style thoughts
`volunteer.case_assignments.any? { |assignment| assignment.active? }` could be `volunteer.case_assignments.active.any?`

`(volunteer.case_contacts.none? ||
          volunteer.case_contacts.maximum(:occurred_at) < (Time.zone.now - 14.days))`
could be just `volunteer.case_contacts.maximum(:occurred_at) < (Time.zone.now - 14.days))`

We might need some `preload` or `includes` on the query to keep from doing n+1 db queries... the rubygem bullet is supposed to detect this for us but sometimes it has a hard time with tests without lots of test data

Reply:
`volunteer.case_assignments.active.any?`

Yeah, duh on my part hah. 

`(volunteer.case_contacts.none? || volunteer.case_contacts.maximum(:occurred_at) < (Time.zone.now - 14.days))`

My concern here is if a volunteer doesn't have any case_contact records, this throws an `undefined for NilClass` error on `<`. I was also thinking in the strictest sense, that if a volunteer has 0 `case_contact` records, then they haven't done anything in 2 weeks. If that's not the intent, then I can just switch to a safe navigator to prevent raising the exception.

Love to hear your thoughts.

Reply:
Good point about the nil error. I was thinking that because it's an array it would just come back nil, but .maximum is not an array. My bad.